### DISCUSSION TITLE
how to set a version starting point

### DISCUSSION THREAD

**Discussion by des2013** [rectorphp/rector#8786] [Q&A] [answered]
I have a large set of php74 code that I need to update to php83.  Simple enough with rector, or so I thought.
Ok, I see how to set the upper limit using this code fragment in rector.php:
   ->withPhpSets([LevelSetList::UP_TO_PHP_83])
But the text discussions say this will start applying rules from php53.  Since the code is already stable running at 74, isn't this a bad way to apply rector?   I do not see any way to set a lower limit.  How do you tell it to start with rules for 74 and move up to 83?  There are no constants defined like START_FROM_PHP_74.  What am I missing here?

**Comment by samsonasik** [comment, marked as answer]
You can define by each list if needed

```php
withSets([
   SetList::PHP_74,
   SetList::PHP_80,
  // etc
])
```

**Reply by des2013** [reply]
So, there should be a constant for each intermediate version. Does the order effect the operation?  The documentation does not show this method. Thanks for the input though.

**Reply by samsonasik** [reply]
The constant already exists, see https://getrector.com/demo/2ab0d4b3-7c1a-4c70-ad11-e2bd442770a0

**Comment by des2013** [comment]
I matched the function declaration arguments, and set only the version flags I wanted:
//withPhpSets(php83, php82, php81, php80, php74, php73, php72, php71, php70, php56, php55, php54, php53, php84)
 ->withPhpSets(true,  true,  true,  true,  true,  false, false, false, false, false, false, false, false, false)
Ran it with
vendor/bin/rector --dry-run --xdebug
and got these messages:
                                                                                                                        
 [WARNING] The "withPhpSets()" method uses named arguments. Its suitable for PHP 8.0+. Use more explicit                
           "withPhp53Sets()" ... "withPhp74Sets()" in lower PHP versions instead.                                       
                                                                                                                        
 [ERROR] Pick only one version target in "withPhpSets()". All rules up to this version will be used.                    
         To use your composer.json PHP version, keep arguments empty. 

The Error message implies that I still have the same problem: it's not starting at 7.4, but 5.3 instead.  Is there no way around this?

**Reply by samsonasik** [reply]
again, use withSets() instead for that for your needed, see https://getrector.com/demo/2ab0d4b3-7c1a-4c70-ad11-e2bd442770a0 that you will only includes rules in the sets defined

**Comment by des2013** [comment]
Tried that before and got errors. Now I see I omitted use line.  ok, seems to be working now.
